### PR TITLE
JCSP-399: DC, Cloud: view projects

### PR DESCRIPTION
**Pull request by rsaliiatlassian**

### CODE REVIEW

File: custom-vu/src/main/kotlin/jces1209/vu/action/ViewProjects.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package jces1209.vu.action
+
+import com.atlassian.performance.tools.jiraactions.api.action.Action
+import com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter
+import jces1209.vu.MeasureType
+import jces1209.vu.page.admin.ViewProjects.ViewProjectsPage
+import net.jcip.annotations.NotThreadSafe
+
+@NotThreadSafe
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/action/ViewProjects.kt]: What is it for?

File: custom-vu/src/main/kotlin/jces1209/vu/action/ViewProjects.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class ViewProjects(
+    private val meter: ActionMeter,
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/action/ViewProjects.kt]: measure

**Review by ysemenkov — CHANGES_REQUESTED** [review]

File: custom-vu/src/main/kotlin/jces1209/vu/page/admin/ViewProjects/CloudViewProjectsPage.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package jces1209.vu.page.admin.ViewProjects
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/admin/ViewProjects/CloudViewProjectsPage.kt]: Let's use the unified format:
1. make lowercase and general name for package `jces1209.vu.page.admin.projects`
2. use Browse for the list and View for single entity - `BrowseProjects`

File: custom-vu/src/main/kotlin/jces1209/vu/page/admin/ViewProjects/CloudViewProjectsPage.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+class CloudViewProjectsPage(
+    private val jira: WebJira
+) : ViewProjectsPage(jira) {
+
+    override val falliblePage = FalliblePage.Builder(
+        jira.driver,
+        and(
+            visibilityOfElementLocated(By.className("aui-page-panel")),
+            visibilityOfElementLocated(By.xpath("//*[@data-test-id = 'global-pages.directories.directory-base.content.table.container']")),
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/admin/ViewProjects/CloudViewProjectsPage.kt]: you can avoid locators duplication by visibilityOfNestedElementsLocatedBy

File: custom-vu/src/main/kotlin/jces1209/vu/page/admin/ViewProjects/CloudViewProjectsPage.kt (hunk 3 above)
Comment (rsaliiatlassian) [reply custom-vu/src/main/kotlin/jces1209/vu/page/admin/ViewProjects/CloudViewProjectsPage.kt]: Let's rename it as Manage projects

image

Comment (rsaliiatlassian) [reply custom-vu/src/main/kotlin/jces1209/vu/page/admin/ViewProjects/CloudViewProjectsPage.kt]: And we already have BrowseProjects class

Comment (rsaliiatlassian) [reply custom-vu/src/main/kotlin/jces1209/vu/page/admin/ViewProjects/CloudViewProjectsPage.kt]: And we already have BrowseProjects class

**Review by ysemenkov — APPROVED** [review]

**Pull request merged**